### ISSUE TITLE
Pass compression options through

### ISSUE THREAD

**Issue by moble** [JuliaDynamics/DrWatson.jl#274]
**Is your feature request related to a problem? Please describe.**
I have some *very* large files (100s of GB) that compress pretty well, and JLD2 supports compression, but I can't use this package's wonderful saving features together with that.

**Describe the solution you'd like**
It would be nice to be able to pass through the `compress=true` option to whatever function is used to actually save the file.  (Maybe ignoring this option for formats that don't support it, which I guess mean JLD and BSON.)  It might be worth keeping in mind more general arguments, as well — for example, if JLD2 gets support for `shuffle`.

**Describe alternatives you've considered**
What I've actually done in my code is to first just write some of the file (a smaller aurxiliary dataset or two) as usual.  This saves all the nice metadata that DrWatson puts together.  I then re-open it with JLD2, and [create a new dataset with the compression option](https://juliaio.github.io/JLD2.jl/dev/compression/#Manually-selecting-compression-for-datasets) I want for the main dataset.  In fact, my production code uses HDF5.jl for access to `shuffle` and compression level.  This means I read most of the file with JLD2, but I have to read the compressed dataset out with HDF5.

**Comment by Datseris** [comment]
Perhaps an even better alternative: allow keyword propagation to the downstream saving function. How do you use `compress`? Can you give it to `FileIO.save`? Can you give some example code?

**Comment by moble** [comment]
Just the plain keyword `compress=true` works.  I hadn't used `FileIO` before (directly), but this simple code does the job:
```julia
using FileIO, CodecZlib, JLD2

x = 0:0.001:2π
y = cis.(x)

FileIO.save("test1.jld2", "x", x, "y", y; compress=true)  # 100604B
FileIO.save("test2.jld2", "x", x, "y", y)  # 106518B
```
(You have to get to somewhat larger sizes for compression to really be worth it, but this gets the point across.)

---

The rest of this is really just a note, and not necessarily actionable:  As for HDF5.jl, it looks like it doesn't accept file-wide compression options.  It looks like its [support for `FileIO` is quite new](https://github.com/JuliaIO/FileIO.jl/releases/tag/v1.10.1), but I can't get it to work with the latest versions.  But for future reference, I wonder if something similar to (but presumably not exactly) this would work:
```julia
FileIO.save("test1.jld2", Dict("x"=>x, ("y", :chunk=>(5,), :compress=>3)=>y))
```
Basically, I wonder if it's possible to emulate the [syntax for creation of a compressed dataset](https://juliaio.github.io/HDF5.jl/stable/#Chunking-and-compression) with the key for the corresponding data.

**Comment by Datseris** [comment]
keyword propagation is quite possible to do. In the end of the day all functions call `wsave`, which itself falls back to `FileIO.save`. So we just have to enable keyword propagation to all `tagsave, safesave, produce_or_load`, etc.

**Comment by moble** [comment]
Closed by #279
